src/inference_result.cc

```diff
@@ -184,32 +185,44 @@ void InferenceResult::show_marginal_snippet(std::ostream &output) const {
   output << "   ..." << std::endl;
 }
 
-void InferenceResult::show_marginal_histogram(std::ostream &output) const {
+void InferenceResult::show_marginal_histogram(std::ostream &output,
+                                              const size_t &bins) const {
```

Comment:
Instead of `const X_t &` I think we should just pass-by-value for these primitive types unless they are `double` or `long long`. See: http://stackoverflow.com/a/14013189/390044